torch/csrc/autograd/variable.h

```diff
@@ -1,89 +1,99 @@
 #pragma once
 
+#include <mutex>
 #include <memory>
 #include <functional>
 #include <THPP/THPP.h>
 
 #include "torch/csrc/autograd/function.h"
-#include "torch/csrc/autograd/saved_variable.h"
+#include "torch/csrc/autograd/variable_version.h"
 #include "torch/csrc/Types.h"
 
 namespace torch { namespace autograd {
 
-struct VariableVersion;
+struct Variable : std::enable_shared_from_this<Variable> {
+
+  struct SavedVariable {
```

Comment:
Isn't the standard C++ practice to forward declare one of the classes in that case? (i.e. `struct Variable;`)